### PR TITLE
Fix NaN in Kubeflow Notebooks

### LINKED ISSUE #5985: Launching notebook from UI using non-default resource requests results in bad spec
/kind bug

**What steps did you take and what happened:**
Attempted to launch a notebook server from the UI with non-default resource requests and it results in NaN values for the resource requests in the spec.

Logs from the controller
```
E0616 02:50:26.491278       1 reflector.go:126] pkg/mod/k8s.io/client-go@v11.0.1-0.20190409021438-1a26190bd76a+incompatible/tools/cache/reflector.go:94: Failed to list *v1beta1.Notebook: v1beta1.NotebookList.Items: []v1beta1.Notebook: v1beta1.Notebook.Spec: v1beta1.NotebookSpec.Template: v1beta1.NotebookTemplateSpec.Spec: v1.PodSpec.Containers: []v1.Container: v1.Container.Resources: v1.ResourceRequirements.Limits: unmarshalerDecoder: quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$', error found in #10 byte of ...|cpu":"NaN","memory":|..., bigger context ...|:"testresource","resources":{"limits":{"cpu":"NaN","memory":"NaNGi"},"requests":{"cpu":"1","memory":|...
```

**What did you expect to happen:**
A valid notebook to be provisioned

**Anything else you would like to add:**
This doesn't happen with the default parameters

**Environment:**

- Kubeflow version: (version number can be found at the bottom left corner of the Kubeflow dashboard): 1.3 https://app.vagrantup.com/arrikto/boxes/minikf/versions/20210428.0.1
- kfctl version: (use `kfctl version`):
- Kubernetes platform: minikf
- Kubernetes version: (use `kubectl version`): v1.16.15
- OS (e.g. from `/etc/os-release`): macOS 11.4

### PR DESCRIPTION
**Pull request by munagekar**
This PR fixes an issue with the Jupyter Web App creating invalid notebook configuration with CPU and Memory set as:

- CPU: NaN
- Memory: NaNGI.

This happens with jupyter-web-app-config when limit factor is set to None and non-default resource value is set.

```yaml
limitFactor: "none"
```
I am not familiar with npm used in the frontend so I am not sure if this is the best way to fix it, hence I sent in a patch to the backend which should fix the issue.

fixes #5985

### CODE REVIEW

File: components/crud-web-apps/jupyter/backend/apps/common/form.py [hunk 1]
```diff
@@ -163,7 +163,10 @@ def set_notebook_cpu(notebook, body, defaults):
     container = notebook["spec"]["template"]["spec"]["containers"][0]
 
     cpu = get_form_value(body, defaults, "cpu")
+
     cpu_limit = get_form_value(body, defaults, "cpuLimit")
+    if cpu_limit == "NaN":
```
Comment (kimwnasptd) [inline components/crud-web-apps/jupyter/backend/apps/common/form.py:168]: let's instead make a check if `NaN` is included in the value of either `cpu` or `cpu_limit`.

And if that's the case then the backend should raise a `BadRequest` werkzeug exception (400)

File: components/crud-web-apps/jupyter/backend/apps/common/form.py [hunk 2]
```diff
@@ -187,7 +190,10 @@ def set_notebook_memory(notebook, body, defaults):
     container = notebook["spec"]["template"]["spec"]["containers"][0]
 
     memory = get_form_value(body, defaults, "memory")
+
     memory_limit = get_form_value(body, defaults, "memoryLimit")
+    if memory_limit == "NaNGi":
```
Comment (kimwnasptd) [inline components/crud-web-apps/jupyter/backend/apps/common/form.py:195]: Same as above for CPU

**Review by kimwnasptd — COMMENTED** [review]
@munagekar following my description in https://github.com/kubeflow/kubeflow/issues/5985#issuecomment-898368072, lets have the backend ensure it doesn't accept malformed values from clients.

File: components/crud-web-apps/jupyter/backend/apps/common/form.py (hunk 1 above)
Comment (munagekar) [reply components/crud-web-apps/jupyter/backend/apps/common/form.py:168]: @kimwnasptd Thank you for the quick review. I will merge in changes from https://github.com/kubeflow/kubeflow/pull/6058 and verify the behavior the locally and make changes as per your suggestions.

**Comment by munagekar** [comment]
@kimwnasptd PTAL, I have made changes as per review suggestion.

**Comment by kimwnasptd** [comment]
The PR looks good, thanks for your time @munagekar!

/ok-to-test
/lgtm

**Comment by kimwnasptd** [comment]
@munagekar there are some formatting issues:
> ./apps/common/form.py:200:80: E501 line too long (84 > 79 characters)

You could use shorter messages like `Invalid value for memory limit: %s`

**Comment by munagekar** [comment]
@kimwnasptd I have fixed lint, PTAL.

**Comment by kimwnasptd** [comment]
Looks good, thanks!

/lgtm

**Pull request merged**